### PR TITLE
crystal: use llvm 10 exclusively

### PR DESCRIPTION
**Pull request by manveru**
###### Motivation for this change

This is trying to fix build failures on darwin

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by manveru** [comment]
So far this has been working well for me, Ideally I'd always like to use the latest version of LLVM, but there's no good alias available.
For now we can update it manually (if I don't forget about it).

Since we always depend on LLVM anyway, I don't think we should use a different stdenv at all.

@peterhoeg

File: pkgs/development/compilers/crystal/default.nix [hunk 1]
```diff
@@ -108,6 +108,8 @@ let
       "all" "docs"
     ];
 
+    LLVM_CONFIG = "${llvm}/bin/llvm-config";
```
Comment (peterhoeg) [inline pkgs/development/compilers/crystal/default.nix:111]: Are you sure this is necessary? Crystal should be handling this for us.

Comment (peterhoeg) [reply pkgs/development/compilers/crystal/default.nix:114]: Ahh, but we still need it for the old versions. Doh.

**Comment by peterhoeg** [comment]
I cannot help with the testing on mac but when #83054 lands, it'll be easy to check all the dependencies on linux at least.

**Comment by manveru** [comment]
@peterhoeg i have access to a few macs for testing this, and it seems perfectly fine so far. Wonder if we can whitelist aarch64 as well now :)

**Comment by peterhoeg** [comment]
>   Wonder if we can whitelist aarch64 as well now :)

Sure, why not.

**Pull request merged**